### ISSUE TITLE
Garbled streaming output under CUDA temperature sampling (temp > 0); non-streaming and greedy are clean

### ISSUE THREAD

**Issue by VictorAnathorn** [timtoole02/Camelid#310]
## Summary

On the CUDA-resident decode path, **streaming** generation produces garbled, off-topic "word-salad" output (often mixed-script) for **any temperature > 0**. The same request is clean when:

- streamed with `temperature: 0` (greedy), or
- requested non-streaming at the same temperature > 0.

So it is specific to **streaming + temperature sampling on the GPU**.

## Reproduction

`camelid serve` a small model on a CUDA GPU (seen with Qwen3 0.6B Q8_0, RTX 5060). Send the same multi-message chat at several temperatures with `stream: true`:

```
temp 0   (greedy) -> "The capital of Chile is Santiago."          (correct)
temp 0.2          -> "ED ??????????f(x) = 2x^2 - 3x + 1 ..."       (garbage)
temp 0.7          -> "Geological ?????? ??? ..."                   (garbage)
temp 1.0          -> "Geological _type_of_knowledge The problem is in a game called Grand Challenges. There are 8 players, each with a watch and a mini-train ..."  (garbage)
```

Non-streaming (`stream: false`) at the same temperatures returns the correct answer every time. The garbage is deterministic-ish (repeats across runs), which points at a specific broken code path rather than ordinary high-temperature variance.

## Cause

Two CUDA-streaming-specific issues:

1. **Resident GPU Gumbel-max temperature sampling.** The streaming decode loop draws sampled tokens via the resident GPU Gumbel-max fast lane (`generate_next_token_sampled_resident` -> `forward_token_sample` -> `sample_gumbel`). That fast lane yields corrupted tokens in the streaming context, whereas greedy/argmax on the same forward is correct (so the forward/logits are fine) and the **CPU** temperature sampler is correct (non-streaming uses it and is clean).

2. **Prompt-prefix cache not bypassed under CUDA in streaming.** The non-streaming path bypasses the prompt-prefix cache when `resident_decode_cuda_active()` (reusing a cached session reseeds GPU KV from f16-rounded host history — a different reduction order than a clean GPU prefill, which flips near-tie tokens). The streaming path uses the cache unconditionally, violating that documented invariant.

## Expected

Streaming temperature sampling on the GPU should produce coherent output, matching non-streaming and greedy.

A fix PR follows: gate the resident GPU temperature-sampling fast lane behind an opt-in env (default off → fall back to the correct CPU sampler), and apply the existing `resident_decode_cuda_active()` cache bypass to the streaming path.
